### ISSUE TITLE
python3.5 - iteritems error

### ISSUE THREAD

**Issue by peterch405** [CGATOxford/UMI-tools#40]
```
Traceback (most recent call last):
  File "/home/chovanec/anaconda3/bin/umi_tools", line 11, in <module>
    sys.exit(main())
  File "/home/chovanec/anaconda3/lib/python3.5/site-packages/umi_tools/umi_tools.py", line 42, in main
    module.main(sys.argv)
  File "/home/chovanec/anaconda3/lib/python3.5/site-packages/umi_tools/dedup.py", line 972, in main
    read_gn = random_read_generator(infile.filename, chrom=options.chrom)
  File "/home/chovanec/anaconda3/lib/python3.5/site-packages/umi_tools/dedup.py", line 763, in __init__
    self.fill()
  File "/home/chovanec/anaconda3/lib/python3.5/site-packages/umi_tools/dedup.py", line 782, in fill
    for observed_umi, freq in self.umis_counter.iteritems():
AttributeError: 'Counter' object has no attribute 'iteritems'
```

Seems like it should be something more along these lines?

```
# Python 2 and 3: option 3
from future.utils import iteritems
# or
from six import iteritems

for (key, value) in iteritems(heights):
    ...
```

http://python-future.org/compatible_idioms.html

**Comment by peterch405** [comment]
Changed line 784-789 in dedup.py:

```
for observed_umi, freq in iteritems(self.umis_counter):
            self.frequency2umis[freq+0.0/total_umis].append(observed_umi)

        self.frequency_counter = collections.Counter(self.umis_counter.values())
        self.frequency_prob = [(float(x)/total_umis)*y for x, y in
                               iteritems(self.frequency_counter)]
```
and managed to break something else, not sure what this time:

```
Traceback (most recent call last):
  File "mtrand.pyx", line 1387, in mtrand.RandomState.choice (numpy/random/mtrand/mtrand.c:15275)
TypeError: 'dict_keys' object cannot be interpreted as an integer
```

**Comment by IanSudbery** [comment]
Looks to me like in Py3, the Counter.keys() function returns something of class dict_keys, where as in py2.7 it returns a list. Wrapping the call to self.frequency_counter.keys() in list() on line 794 might do the trick.

**Comment by peterch405** [comment]
Created a pull request which hopefully fixes this.

**Comment by IanSudbery** [comment]
Merged PR #41 . 

Thanks for your help.
Can we close the issue?

**Comment by peterch405** [comment]
I would say yes. 
Thank you.
